### DISCUSSION TITLE
Can´t pair any devices

### DISCUSSION THREAD

**Discussion by workinganimal** [Koenkk/zigbee2mqtt#11361] [General]
Hi!

My problem started with one motion sensor RTCGQ01LM not detecting movement in the usual time or not at all. So I thought maybe the batteries were dead and changed them. After that it didn't do anything. So i tried repairing. It did not pair. Then i took another sensor RTCGQ11LM. Also don't work. Gave them to my friend who confirmed that his system pairs them normally. Then I thought maybe my Zigbee adapter Texas Instruments LAUNCHXL-CC1352P-1 is the problem. Got one workin unit from a friend and changed the ieee and so on and still the same. Now I have tried different devices and nothing is pairing. I have about 35 xiaomi aqara devices working but if I try to repair any of them then they will not work anymore. I think that the problem came with the last zigbee2mqtt update but I am not sure and I don´t have a backup old enough to try to downgrade. 

I have tried several days to figure out my problem but nothing has helped so far. Also I have noticed that zigbee devices are slow. Beside xiaomi devices i have 5x osram spotlights 4058075148338 and a gledopto ledstrip dimmer. I removed all the spotlights from my installation so that they can´t be routers. Alos they don´t have power all the time so this didn't help.

What should I do. I have a lot of automations and devices and really don't want to do things from the beginning. I really hope that someone can help me with this.

Best regards.

**Comment by phretor** [comment]
I'm facing the exact same problem. Started with a cheap CC2531 and 2-3 IKEA routers: low LQI, but overall all my Aqara motion sensors, contacts, and temp sensors have worked well for about 6-8 months. Never had to replace batteries.

2 months ago I installed 12 Zemismart 3-gang switches to build a solid mesh. Worked well with Z2M but I started see the limits of the CC2531. So I got a slaesh which came preflashed with Z-Stack 3.x.0. I built a solid mesh on channel 15, but then I started to notice that all Aqara nodes would drop after 8-12hrs. After some digging I found out that they don't play well on channel 15, so I decided to switch to channel 11 or 20.

In the meanwhile, I think, Z2M got upgraded (I use it via the HASS addon so it upgrades almost automatically).

That's when I started to be unable to pair ANY device, no matter whether they're battery or mains powered. Anyhow, I tested and replaced ALL batteries (>=3.3V). I tried to pair at various distances from the coordinator, uses the antenna that came with the slaesh dongle, which worked well, USB extender, etc.

I reflashed the dongle with the latest firmware version (and the previous one), always flashed fine, but nothing has changed.

A somewhat constant element I noticed is that, even if I switch off all mains (to ensure they're not broadcasting the old network info), at the first-second boot, Z2M complains that there may be another network with the same pan/ext ID, **even if** I change all network details at every reboot, and remove the zigbee2mqtt folder altogether.

Only yesterday, and it took it 200 seconds, I managed to pair a Zemismart switch on channel 20, but it turned out to be completely unreliable (I wasn't able to turn on the switch via ZigBee, but I could see the state changing). I think that was almost as an accident, as I wasn't able to pair anything else.

Anything we should try @Koenkk?

I turned logging all the way to DEBUG level, but didn't notice anything unexpected (except the pan/ext ID warnings I mentioned above). The stock's LED turns blue when in "permit join" mode. The stick passes the [coordinator test](https://slae.sh/projects/cc2652/#checking-the-functionality-and-reading-fw-version) with both Z-Stack and Slae.sh firmware.

I have scopes (SDRs) if you want me to take any measurements to help you debug what may be happening.

**Reply by Koenkk** [reply]
> but didn't notice anything unexpected (except the pan/ext ID warnings I mentioned above).

These can happen when the fw hasn't been flashed correctly, try flashing on a different system: https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#error-coordinator-failed-to-start-probably-the-panid-is-already-in-use-try-a-different-panid-or-channel

**Reply by phretor** [reply]
> These can happen when the fw hasn't been flashed correctly, try flashing on a different system: https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html#error-coordinator-failed-to-start-probably-the-panid-is-already-in-use-try-a-different-panid-or-channel

Would this imply that we should not rely on the flashing program's feedback? Here's what I mean with an example:

```bash
$ python cc2538-bsl.py -evw ../Z-Stack-firmware/coordinator/Z-Stack_3.x.0/bin/CC2652RB_coordinator_20211217/CC2652RB_coordinator_20211217.hex
Opening port /dev/tty.usbserial-1410, baud 500000
Reading data from ../Z-Stack-firmware/coordinator/Z-Stack_3.x.0/bin/CC2652RB_coordinator_20211217/CC2652RB_coordinator_20211217.hex
Your firmware looks like an Intel Hex file
Connecting to target...
CC1350 PG2.0 (7x7mm): 352KB Flash, 20KB SRAM, CCFG.BL_CONFIG at 0x00057FD8
Primary IEEE Address: 00:12:4B:00:23:93:41:1C
    Performing mass erase
Erasing all main bank flash sectors
    Erase done
Writing 360448 bytes starting at address 0x00000000
Write 104 bytes at 0x00057F980
    Write done
Verifying by comparing CRC32 calculations.
    Verified (match: 0x49cb5a2e)
```

I take this as "flashed correctly". But what you're telling me now is that there's still room for error, despite the CRC32 matching?

**Reply by Koenkk** [reply]
>  But what you're telling me now is that there's still room for error, despite the CRC32 matching?

Exactly, it is very strange but someone made a dump of the stick when flashed with CC2538 vs e.g. Uniflash and the contents differ.

**Reply by phretor** [reply]
> Exactly, it is very strange but someone made a dump of the stick when flashed with CC2538 vs e.g. Uniflash and the contents differ.

Good to know. At some point there should be an HASS add-on for flashing :-) for now I'll stick to a random Linux box I have lying around.

Thanks!

**Comment by phretor** [comment]
For the records, re-flashing from a Linux box solved the problem.

**Reply by workinganimal** [reply]
Can you give me any advice how to do it in Linux. I don´t have a lot of experience with Linux but i can run some distro from a flash drive or create a vm in my Unraid server but how to i flash the coordinator firmware in Linux?

**Reply by workinganimal** [reply]
Sorry for the dumb question. There are instructions on zigbee2mqtt site :) My bad! I tried on different windows machines with no luck. And this is really weird because on one machine i have made several firmware flashes and it has always worked till now. I will try to flash it on linux today and i really hope that it will work again.

**Reply by workinganimal** [reply]
I still haven´t had any luck. Yesterday i flashed the adapter on a Ubuntu vm with cc2538-bsl and nothing changed. For me this is weird because my problem started before re flashing the adapter firmware. I am really frustrated about this situation. I have used it for about 5 years and now this :( . I will try an new installation with my rpi4 and another adapter and see what happens but i am sure that the problem remains... Is it possible to get older coordinator firmware for Texas Instruments LAUNCHXL-CC1352P?

**Reply by workinganimal** [reply]
I have tried on 6 different machines without luck. Is it possible that the new coordinator firmware does not suit for the Texas Instruments LAUNCHXL-CC1352P1?

**Comment by workinganimal** [comment]
So now i am sure that the problem is not with my adapter. I borrowed a conbee II adabter for testing. Made a new installation with zha and still the same thing. Can´t pair any devices. I am really stuck here. Could someone plese give me some advice? What should I try to do?

**Comment by workinganimal** [comment]
Now i got it working with a conbee II stick but still i would like to use my Texas Instruments LAUNCHXL-CC1352P-1
@Koenkk Can i download older firmware for my adapter somewhere? This has to be firmware related because i have tried so many different systems. Also all the systems where it worked about 2 years ago.

**Comment by workinganimal** [comment]
@Koenkk I found a solution! The latest firmware does not suit for Texas Instruments LAUNCHXL-CC1352P-1 !
I got the Z-Stack_3.x.0 20210120 firmware and everything works without problems again. Can you do anything about that or are these boards useless with new firmware?

**Reply by Koenkk** [reply]
It was never the intention for these fws to work on a CC1352P1, I only support the CC1352P2.

**Reply by workinganimal** [reply]
That´s too bad! It is faster and has a much better range than the conbee II stick. When I started with zigbee2mqtt about 5 years ago then i asked you about this board and you didn´t see a problem why this board should not work. Is it possible that it would work with the latest firmware if i set the transmit power to 5dBm that is the maximum of CC1352P1?

**Reply by workinganimal** [reply]
It works. I flashed the CC1352P2_CC2652P_launchpad_coordinator_20220219.hex and added the following to configuration:
```
experimental:
  transmit_power: 5
```
